### ISSUE TITLE
How can I get intermediate weights/gradients/layer_outputs while training?

### ISSUE THREAD

**Issue by biofoolgreen** [TensorSpeech/TensorFlowTTS#656]
Hi, I'm debugging a numerial issue on FastSpeech2. I'd like to printing/saving weights/gradients/outputs of each layer. I've tried add logger to `_calculate_gradient_per_batch` function in `base_trainer.py` but only got tensors not numpy values.

How could I get intermediate weights/gradients of each layer while training? It's very appreciated if you have any convienient way to debug numerical issue. Thanks in advance.

**Comment by dathudeptrai** [comment]
can you tell me what exactly a bug is?

**Comment by EmreOzkose** [comment]
@biofoolgreen did you reach mid-layers? I wanna reach too, to just observe.

**Comment by EmreOzkose** [comment]
we can do something like that for multiband-melgan:
```
intermadiate_layer_index = 20 (the layer we want)
feature_extactor = mb_melgan.get_layer(index=0)
inter_output_model = tf.keras.Model(feature_extactor.input, feature_extactor.get_layer(index = intermadiate_layer_index).output)

feat = mel_spectrogram    # (1, #frame, 80)
intermediate_output = inter_output_model(feat)    # (1, #frame, #feat_dimension)
```
